backend/controllers/permission_controller.py

```diff
@@ -0,0 +1,175 @@
+from db import get_db
+from db.orm import Category, Page, User, UserCategoryPermission, UserPagePermission
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from controllers.dependencies import get_admin_user, get_current_user
+
+PERMISSION_CONTROLLER = APIRouter(prefix="/permissions")
+
+
+@PERMISSION_CONTROLLER.post("/")
+def add_permission(
+    user_id: int,
+    page_id: int,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_admin_user),
+):
+    """Add permission for editor to access specific page"""
+    user = db.query(User).filter(User.id == user_id, User.role == 1).first()
+    if not user:
+        raise HTTPException(status_code=404, detail="Editor not found")
+
+    page = db.query(Page).filter(Page.id == page_id).first()
+    if not page:
+        raise HTTPException(status_code=404, detail="Page not found")
+
+    existing = (
+        db.query(UserPagePermission).filter_by(user_id=user_id, page_id=page_id).first()
+    )
+    if existing:
+        raise HTTPException(status_code=400, detail="Permission already exists")
+
+    new_permission = UserPagePermission(user_id=user_id, page_id=page_id)
+    db.add(new_permission)
+    db.commit()
+    return {"message": "Permission added"}
+
+
+@PERMISSION_CONTROLLER.delete("/")
+def remove_permission(
+    user_id: int,
+    page_id: int,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_admin_user),
+):
+    """Remove permission from editor for specific page"""
+    permission = (
+        db.query(UserPagePermission).filter_by(user_id=user_id, page_id=page_id).first()
+    )
+    if not permission:
+        raise HTTPException(status_code=404, detail="Permission not found")
+
+    db.delete(permission)
+    db.commit()
+    return {"message": "Permission removed"}
+
+
+@PERMISSION_CONTROLLER.get("/{user_id}/pages")
+def get_user_permissions(
+    user_id: int,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user),
+):
+    """Get list of page IDs that user has permission to edit"""
+    if current_user.role != 2 and current_user.id != user_id:
+        raise HTTPException(status_code=403, detail="Access denied")
+
+    permissions = db.query(UserPagePermission.page_id).filter_by(user_id=user_id).all()
+    return [p.page_id for p in permissions]
+
+
+@PERMISSION_CONTROLLER.get("/{user_id}/pages/{page_id}")
+def check_user_permission(
+    user_id: int,
+    page_id: int,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user),
+):
+    """Check if user has permission to edit specific page"""
+
+    if current_user.role == 2:
+        return {"has_permission": True}
+
+    permission = (
+        db.query(UserPagePermission).filter_by(user_id=user_id, page_id=page_id).first()
+    )
+
+    return {"has_permission": permission is not None}
```

Comment:
Môžeš vysvetliť logiku toho ako toto funguje, prosím?

Frontend odošle na tento endpoint požiadavku, server povie či má na nejakú page permission a frontend to potom rešpektuje? Prebieha aj kontrola permission pri endpointe ktorý edituje samotnú stránku v databáze?